### PR TITLE
[Metrics API] Update Instrument Provider and remove `try_init` methods

### PR DESCRIPTION
**Pull request by utpilla**
Towards #1699

## Changes
- Update `InstrumentProvider` trait methods to return the individual instruments instead of a `Result`
- Remove `try_init` methods from the all the instrument builders

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry/src/metrics/instruments/mod.rs [hunk 1]
```diff
@@ -95,34 +95,24 @@ impl<'a, T> HistogramBuilder<'a, T> {
...
-        self.instrument_provider.f64_histogram(self)
-    }
-
     /// Creates a new instrument.
     ///
     /// Validates the instrument configuration and creates a new instrument. In
-    /// case of invalid configuration, an instrument that is no-op is returned
+    /// case of invalid configuration, a no-op instrument is returned
     /// and an error is logged using internal logging.
     pub fn init(self) -> Histogram<f64> {
```
Comment (lalitb) [inline opentelemetry/src/metrics/instruments/mod.rs:103]: I believe this `init` method would be renamed to `build` as separate PR, as discussed in community meeting today?

File: opentelemetry-sdk/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 - Improved `LoggerProvider` shutdown handling to prevent redundant shutdown calls when `drop` is invoked. [#2195](https://github.com/open-telemetry/opentelemetry-rust/pull/2195)
 - **BREAKING**: [#2217](https://github.com/open-telemetry/opentelemetry-rust/pull/2217)
   - **Replaced**: Removed `{Delta,Cumulative}TemporalitySelector::new()` in favor of directly using `Temporality` enum to simplify the configuration of MetricExporterBuilder with different temporalities.
+- When creating new metric instruments, SDK would return a no-op instrument if the validation fails. [#2166](https://github.com/open-telemetry/opentelemetry-rust/pull/2166)
```
Comment (lalitb) [inline opentelemetry-sdk/CHANGELOG.md:10]: nit - PR link and number.

**Review by lalitb — APPROVED** [review]

File: opentelemetry/src/metrics/instruments/mod.rs (hunk 1 above)
Comment (utpilla) [reply opentelemetry/src/metrics/instruments/mod.rs:103]: Yes. I'll do that in a follow-up PR.

File: opentelemetry-sdk/CHANGELOG.md (hunk 2 above)
Comment (utpilla) [reply opentelemetry-sdk/CHANGELOG.md:10]: This particular change of SDK returning a no-op instrument if validation fails was actually covered in an earlier PR. We had not updated the CHANGELOG back then. Check this https://github.com/open-telemetry/opentelemetry-rust/pull/2166#discussion_r1785181584

**Review by cijothomas — APPROVED** [review]

**Pull request merged**